integration/test_ml.py

```diff
@@ -0,0 +1,361 @@
+from tests import testutils
+
+
+# pylint: disable=import-error,no-name-in-module
+try:
+    import tensorflow as tf
+    _TF_ENABLED = True
+except ImportError:
+    _TF_ENABLED = False
+
+
+NAME_ONLY_ARGS = {
+    'display_name': 'TestModel123_{0}'.format(random.randint(1111, 9999))
+}
+NAME_ONLY_ARGS_UPDATED = {
+    'display_name': 'TestModel123_updated_{0}'.format(random.randint(1111, 9999))
+}
+NAME_AND_TAGS_ARGS = {
+    'display_name': 'TestModel123_tags_{0}'.format(random.randint(1111, 9999)),
+    'tags': ['test_tag123']
+}
+FULL_MODEL_ARGS = {
+    'display_name': 'TestModel123_full_{0}'.format(random.randint(1111, 9999)),
+    'tags': ['test_tag567'],
+    'file_name': 'model1.tflite'
+}
+INVALID_FULL_MODEL_ARGS = {
+    'display_name': 'TestModel123_invalid_full_{0}'.format(random.randint(1111, 9999)),
+    'tags': ['test_tag890'],
+    'file_name': 'invalid_model.tflite'
+}
+
+@pytest.fixture
+def firebase_model(request):
+    args = request.param
+    tflite_format = None
+    file_name = args.get('file_name')
+    if file_name:
+        file_path = testutils.resource_filename(file_name)
+        source = ml.TFLiteGCSModelSource.from_tflite_model_file(file_path)
+        tflite_format = ml.TFLiteFormat(model_source=source)
+
+    ml_model = ml.Model(
+        display_name=args.get('display_name'),
+        tags=args.get('tags'),
+        model_format=tflite_format)
+    model = ml.create_model(model=ml_model)
+    yield model
+    _clean_up_model(model)
+
+
+@pytest.fixture
+def model_list():
+    ml_model_1 = ml.Model(display_name="TestModel123")
+    model_1 = ml.create_model(model=ml_model_1)
+
+    ml_model_2 = ml.Model(display_name="TestModel123_tags", tags=['test_tag123'])
+    model_2 = ml.create_model(model=ml_model_2)
+
+    yield [model_1, model_2]
+
+    _clean_up_model(model_1)
+    _clean_up_model(model_2)
+
+
+def _clean_up_model(model):
+    try:
+        # Try to delete the model.
+        # Some tests delete the model as part of the test.
+        ml.delete_model(model.model_id)
+    except exceptions.NotFoundError:
+        pass
+
+
+# For rpc errors
+def check_firebase_error(excinfo, status, msg):
+    err = excinfo.value
+    assert isinstance(err, exceptions.FirebaseError)
+    assert err.cause is not None
+    assert err.http_response is not None
+    assert err.http_response.status_code == status
+    assert str(err) == msg
+
+
+# For operation errors
+def check_operation_error(excinfo, msg):
+    err = excinfo.value
+    assert isinstance(err, exceptions.FirebaseError)
+    assert str(err) == msg
+
+
+def check_model(model, args):
+    assert model.display_name == args.get('display_name')
+    assert model.tags == args.get('tags')
+    assert model.model_id is not None
+    assert model.create_time is not None
+    assert model.update_time is not None
+    assert model.locked is False
+    assert model.etag is not None
+
+
+def check_model_format(model, has_model_format, validation_error):
```

Comment:
Set `has_model_format=False, validation_error=None` and simplify the call-site.

Reply:
done